### DISCUSSION TITLE
[New Feature] Harvester=yes units auto-requesting Carryall pickup

### DISCUSSION THREAD

**Discussion by TAK02** [Phobos-developers/Phobos#946] [Ideas and feedback]
### Description:
In Dune 2, Dune 2000 and Emperor: Battle for Dune Harvester units automatically request pickup from Carryall units.
Would love to see this some day.

### Possible Implementation:
A system that makes `Harvester=yes` units request a `Carryall=yes` unit when going anywhere, that carryall automatically picks up the Harvester and carries over the unit's move order into its own so the Harvester is set down at that cell.

Of course, it'll have to be like the Subterranean Locomotor in a sense: only call for Carryall if the target cell will need a path longer than ~10 cells or something, don't know the exact number. I think it was actually 11 cells for the Subterranean Locomotor before the unit would dig/consider digging instead of driving?

Here's what I'm not sure about what happens in the Dune games: when a carryall is ordered to pick up a unit, does the unit stop or does the unit keep going until it gets picked up?
Don't know how the "advanced"/player-controlled Carryalls do it in EBfD, but I do know how the non-player controlled Harvester Carryalls can pick up moving harvesters in EBfD.
No idea how things are handled in Dune 2000 or Dune 2.

**Comment by ghost** [comment]
In Dune 2 harvesters request pickup whenever they are full, regardless of distance to base, then they stop and wait for pickup. They only walk back if there's no carryall available. If they start walking, they will no longer request pickup until full again. If the harvester was returned to base by a carryall, it will be returned to the last harvesting spot after unload if a carryall is still available to do so.

**Comment by ajgunsauley** [comment]
It would be cool to generalize this such that it's not dependent on a harvester and could be extended to other vehicles as well.  I know of at least one unit design where the intent was to have the vehicle be basically completely immobile and dependent on carryalls.  In that case, having a tag that's something like LookForCarryalls=Yes would have a carryall try to fly it whenever a move order for the base unit was given.
